python/test_gilded_rose.py

```diff
@@ -5,12 +5,153 @@
 
 
 class GildedRoseTest(unittest.TestCase):
-    def test_foo(self):
-        items = [Item("foo", 0, 0)]
-        gilded_rose = GildedRose(items)
-        gilded_rose.update_quality()
-        self.assertEqual("fixme", items[0].name)
+	def test_foo(self):
+		items = [Item("foo", 0, 0)]
+		gilded_rose = GildedRose(items)
+		gilded_rose.update_quality()
+		self.assertEqual("foo", items[0].name)
+
+	def test_items_correctly_added(self):
+		items = [Item("foo", 0, 0), Item("bar", 0, 0),
+				 Item("fizz", 0, 0), Item("buzz", 0, 0),
+				 Item("bing", 0, 0), Item("bang", 0, 0),
+				 Item("bong", 0, 0)]
+		gilded_rose = GildedRose(items)
+		gilded_rose.update_quality()
+		self.assertEqual("foo", items[0].name)
+		self.assertEqual("bar", items[1].name)
+		self.assertEqual("fizz", items[2].name)
+		self.assertEqual("buzz", items[3].name)
+		self.assertEqual("bing", items[4].name)
+		self.assertEqual("bang", items[5].name)
+		self.assertEqual("bong", items[6].name)
+
+	def test_a_lot_of_items_correctly_added(self):
+		items = []
+		for i in range(10000):
+			items.append(Item(str(i), 0, 0))
+		gilded_rose = GildedRose(items)
+		gilded_rose.update_quality()
+		for i in range(10000):
+			self.assertEqual(str(i), items[i].name)
+
+	def test_update_quality_once(self):
+		items = [Item("Aged Brie", 100, 30),
+				 Item("Sulfuras, Hand of Ragnaros", 30, 80), Item("Chips", 5, 10)]
+		gilded_rose = GildedRose(items)
+		gilded_rose.update_quality()
+		self.assertEqual("Aged Brie", items[0].name)
+		self.assertEqual(99, items[0].sell_in)
+		self.assertEqual(31, items[0].quality)
+		self.assertEqual("Sulfuras, Hand of Ragnaros", items[1].name)
+		self.assertEqual(30, items[1].sell_in)
+		self.assertEqual(80, items[1].quality)
+		self.assertEqual("Chips", items[2].name)
+		self.assertEqual(4, items[2].sell_in)
+		self.assertEqual(9, items[2].quality)
+
+	def test_update_quality_multiple(self):
+		items = [Item("Aged Brie", 100, 30),
+				 Item("Sulfuras, Hand of Ragnaros", 30, 80), Item("Chips", 5, 10)]
+		gilded_rose = GildedRose(items)
+		for i in range(10):
+			gilded_rose.update_quality()
+		self.assertEqual("Aged Brie", items[0].name)
+		self.assertEqual(90, items[0].sell_in)
+		self.assertEqual(40, items[0].quality)
+		self.assertEqual("Sulfuras, Hand of Ragnaros", items[1].name)
+		self.assertEqual(30, items[1].sell_in)
+		self.assertEqual(80, items[1].quality)
+		self.assertEqual("Chips", items[2].name)
+		self.assertEqual(-5, items[2].sell_in)
```

Comment:
I think we should cap the `sell_in` date at `0` - we shouldn't have negative `sell_in` or `quality`